src/iterative_ensemble_smoother/_iterative_ensemble_smoother.py

```diff
@@ -123,8 +124,22 @@ def fit(
         if noise is None:
             noise = rng.standard_normal(size=(num_obs, ensemble_size))
 
-        E = make_E(observation_errors, noise)
+        # Columns of E should be sampled from N(0,Cdd) and centered, Evensen 2019
+        if R is not None:
+            # Assuming R and observation_errors has relation
```

Comment:
Mind explaining this in a bit more detail, as I don't quite see why it's necessary to multiply `R` by the variance of the observation_errors.

Reply:
I thought it would be like this due to `_create_errors()`

```py
def _create_errors(
    observation_errors: npt.NDArray[np.double],
    inversion: InversionType,
) -> Tuple[npt.NDArray[np.double], npt.NDArray[np.double]]:
    if len(observation_errors.shape) == 2:
        R = observation_errors
        observation_errors = np.sqrt(observation_errors.diagonal())
        R = (R.T / R.diagonal()).T
    elif len(observation_errors.shape) == 1 and inversion == InversionType.EXACT_R:
        R = np.identity(len(observation_errors))
    else:
        R = None
    return R, observation_errors
```

that is called immediately in `SIES.fit()`.
The way I read it,  if `observation_errors` is passed as a covariance matrix (which is then `C_dd`, right, or am I thinking wrongly on notation?), then a scaling with the variance is performed, and the scaled covariance matrix (now `R`) and the standard deviations (root diagonal) are returned. 

If `observation_errors` is passed as a vector with standard deviations (root diagonal of `C_dd`), then `R` is taken as the identity matrix. 

In both of these cases I try to recreate `C_dd`.

Reply:
`observation_errors` is either a 1D array of standard deviatons or a 2D covariance matrix. Not the best API perhaps.
Then I think we can do something like the following:

```python
if len(observation_errors.shape) == 2:
    E = np.linalg.cholesky(observation_errors) @ noise
else:
    E = np.linalg.cholesky(np.diag(observation_errors ** 2)) @ noise
``` 

May be wrong though!

Reply:
Because we call
```py
R, observation_errors = _create_errors(observation_errors, inversion)
```
`observation_errors` is always modified. In the case when it is initially passed as a 2D covariance matrix it is modified here
```
observation_errors = np.sqrt(observation_errors.diagonal())
```
The way I read it, is it not always modified by `_create_errors` to be a 1D array of standard deviations?

Reply:
Perhaps we can make E before calling `_create_errors`?

Reply:
Would this not require the code to be async?

Reply:
I think I will modify `_create_errors`, and output also `E` from this function. It makes sense considering the naming as well.

Reply:
Tried creating `E` before everything else. This requires `noise`, which requires `ensemble_size`, which must be validated first.
If not it breaks test 
`test_that_bad_inputs_cause_nice_error_messages():`

I am thinking to modify `_validate_inputs` to handle `observation_errors` as optional between 1D and 2D arrays, and place the validation first in `SIES.fit`.

Then we may create `noise`, `E`, and `R` in that order.